src/components/HeatMap/Cell/Cell.js

```diff
@@ -1,15 +1,19 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { setActiveCell, setHourPosts } from '../../../redux/actions/actions';
 import { Wrapper } from './Cell.styles';
 
 
-export default function Cell({
-  itemsNumber, weekDay, hour, activeCell, onCellClick,
+function Cell({
+// eslint-disable-next-line no-shadow
+  itemsNumber, weekDay, hour, activeCell, setActiveCell, hourData, setHourPosts,
 }) {
   const isActive = activeCell[0] === weekDay && activeCell[1] === hour;
 
   const handleClick = () => {
-    onCellClick(weekDay, hour);
+    setActiveCell([weekDay, hour]);
+    setHourPosts(hourData);
```

Comment:
Here you store the post for the posts for the selected hour inside the heatmap, right? This way you duplicate the posts in your state. Once they are stored in the data state and once in the hourPosts.

You don't need to do that because you have the active cell. So you can simply get the data when you render the table with something like `data[activeCell[0]][activeCell[1]]`.

The important part to learn here is having a "single source of truth". This should be your data state.

Reply:
I was aware of that. But what I store in state is the raw data. The sorted data is just a useMemo function. I was thinking if it wouldn't be better to sotore it instead of the data - but then it happens not at the top level of the app.

Reply:
Actually my idea was to store the sorted data ( when I find how ) and then I wouldn't need to map the data down throw the  heatmap and the row but filter it directly in the cell as you suggest.

Reply:
I would move the business logic for mapping the response to the actual posts data to the reducer. That's where you usually find the heavier logic